backend/src/db/repositories/UserRepository.ts

```diff
@@ -261,4 +262,11 @@ export default class UserRepository {
             gender
         });
     }
+
+    async getUsernameSuggestions(start: string): Promise<string[]> {
+        return await this.db.fetchAll<string>(
```

Comment:
That probably works as a first version, but it's better to cache usernames in memory, ideally in [a Trie](https://en.wikipedia.org/wiki/Trie).